### PR TITLE
Preserve original page objects when region remains unchanged

### PR DESCRIPTION
**Pull request by Akanksha-kedia**
## Description
Optimized  code to preserve the original page objects when the region remains unchanged. This means that new objects will no longer be created unnecessarily when the region is the same as the original page. 

## Motivation and Context
 By avoiding the creation of new objects when the region is the same as the original page, we can reduce memory usage

## Impact
NO MAJOR IMPACT, PRESTO-COMMON MODULE

## Test Plan
Unit tests to verify that new objects are not created when the region remains unchanged in presto-common module.
[INFO] Tests run: 261, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.106 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 261, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  41.197 s
[INFO] Finished at: 2023-12-15T14:43:24+05:30
[INFO] ------------------------------------------------------------------------

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

== NO RELEASE NOTE ==

### CODE REVIEW

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please review.

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please review.

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Please ALWAYS attribute changes from other projects.

File: presto-common/src/test/java/com/facebook/presto/common/TestPage.java [hunk 1]
```diff
@@ -40,14 +40,17 @@ public class TestPage
     @Test
     public void testGetRegion()
     {
-        assertEquals(new Page(10).getRegion(5, 5).getPositionCount(), 5);
+        Page page = new Page(10);
+        assertEquals(page.getRegion(5, 5).getPositionCount(), 5);
+        assertEquals(page.getRegion(0, 10).getPositionCount(), 10);
```
Comment (tdcmeehan) [inline presto-common/src/test/java/com/facebook/presto/common/TestPage.java:45]: Can you please add a test that shows that this is the same object?  I.e. `page.getRegion(0, 10)` === `page`

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

**Comment by tdcmeehan** [comment]
Still not addressed--use `assertSame`

File: presto-common/src/test/java/com/facebook/presto/common/TestPage.java (hunk 1 above)
Comment (Akanksha-kedia) [reply presto-common/src/test/java/com/facebook/presto/common/TestPage.java]: have added Page page = new Page(10);
        Page region = page.getRegion(0, 10);
        assertEquals(page.getRegion(5, 5).getPositionCount(), 5);
        assertEquals(region.getPositionCount(), 10);
        assertSame(page, region);

**Comment by tdcmeehan** [comment]
Use `org.testng.Assert.assertSame`

**Pull request closed without merging**